src/rdkafka_cgrp.c

```diff
@@ -3420,6 +4702,93 @@ static void rd_kafka_cgrp_timeout_scan (rd_kafka_cgrp_t *rkcg, rd_ts_t now) {
 }
 
 
+/**
+ * @brief Handle an assign op.
+ * @locality rdkafka main thread
+ * @locks none
+ */
+static void rd_kafka_cgrp_handle_assign_op (rd_kafka_cgrp_t *rkcg,
+                                            rd_kafka_op_t *rko) {
+        rd_kafka_resp_err_t err = RD_KAFKA_RESP_ERR_NO_ERROR;
+        rd_kafka_error_t *error = NULL;
+
+        if (rkcg->rkcg_flags & RD_KAFKA_CGRP_F_TERMINATE) {
+
+                /* Check that the op is valid for the terminating case */
+                if ((rko->rko_u.assign.method ==
+                     RD_KAFKA_ASSIGN_METHOD_ASSIGN ||
+                     rko->rko_u.assign.method ==
+                     RD_KAFKA_ASSIGN_METHOD_INCR_ASSIGN ||
+                     rkcg->rkcg_assignment == NULL) &&
+                    rko->rko_u.assign.partitions)
+                        err = RD_KAFKA_RESP_ERR__DESTROY;
+                else if (rko->rko_u.assign.method ==
+                         RD_KAFKA_ASSIGN_METHOD_INCR_UNASSIGN &&
+                         rko->rko_u.assign.partitions->cnt !=
+                         rkcg->rkcg_assignment->cnt)
+                        err = RD_KAFKA_RESP_ERR__DESTROY;
+                else if (rko->rko_u.assign.method ==
+                         RD_KAFKA_ASSIGN_METHOD_INCR_UNASSIGN) {
+                        int i;
+                        rd_kafka_topic_partition_list_sort_by_topic(
```

Comment:
Add a comment what we're trying to achieve here.

Reply:
sure